src/main/java/uk/gov/justice/probation/courtcaseservice/service/CustodyService.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.probation.courtcaseservice.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.annotation.RequestScope;
+import reactor.core.publisher.Mono;
+import uk.gov.justice.probation.courtcaseservice.restclient.ConvictionRestClient;
+import uk.gov.justice.probation.courtcaseservice.restclient.CustodyRestClient;
+import uk.gov.justice.probation.courtcaseservice.restclient.OffenderRestClient;
+import uk.gov.justice.probation.courtcaseservice.restclient.OffenderRestClientFactory;
+import uk.gov.justice.probation.courtcaseservice.restclient.communityapi.model.CommunityApiOffenderResponse;
+import uk.gov.justice.probation.courtcaseservice.restclient.communityapi.model.OtherIds;
+import uk.gov.justice.probation.courtcaseservice.service.exceptions.CustodyNotFoundException;
+import uk.gov.justice.probation.courtcaseservice.service.exceptions.ExpectedCustodyNotFoundException;
+import uk.gov.justice.probation.courtcaseservice.service.exceptions.NomsNumberNotAvailableException;
+import uk.gov.justice.probation.courtcaseservice.service.model.Conviction;
+import uk.gov.justice.probation.courtcaseservice.service.model.Custody;
+import uk.gov.justice.probation.courtcaseservice.service.model.KeyValue;
+
+@AllArgsConstructor
+@RequestScope
```

Comment:
Just wondering why we have request scope here. I have used this before with the controller but not so familiar with using it on service tier. From what I read  , "The request scope creates a bean instance for a single HTTP request, while the session scope creates a bean instance for an HTTP Session."  See section 4 of this https://www.baeldung.com/spring-bean-scopes.

Reply:
This is to allow the oauth2 to dynamically add the username to the community-api oauth token for its exclusion checks. There's an explanatory comment [here](https://github.com/ministryofjustice/court-case-service/blob/19164b1e6e5f6bff47a4f03a0bd8f96085d3d036/src/main/java/uk/gov/justice/probation/courtcaseservice/restclient/OffenderRestClientFactory.java#L62) but the gist of it is that the Service has to be created per-request and injected with a community-api webclient which has been pre-seeded with the current username. 

Without the request scope it'll be instantiated once on the first request or at startup (can't recall which) and will from then on have an empty or incorrect value for username.

As to using it on the service tier, it shouldn't matter really where it's used as long as it's done on the same thread. Spring creates a proxy object which gets injected in place of the instance to the controller, so it' agnostic to the fact that the service is RequestScoped. 

It's quite a complex setup so I'm happy to walk you through the code at some point if you like, I should really get an ADR raised as well as it's decidedly non-obvious. I'll raise a task to do that.